sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -278,7 +278,7 @@ case class GetMapValue(child: Expression, key: Expression)
       }
     }
 
-    if (!found) {
+    if (!found || map.valueArray().isNullAt(i)) {
```

Comment:
just realized the `GetArrayItem` also need to add null check for interpreted version `eval`, can you also fix that please?